src/server/services/adoptable-trash-containers.ts

```diff
@@ -0,0 +1,139 @@
+import { differenceInYears } from 'date-fns';
+import { LatLngBoundsLiteral } from 'leaflet';
+
+import { fetchBRP } from './brp';
+import { fetchDataset } from './buurt/buurt';
+import {
+  datasetEndpoints,
+  DatasetFeatureProperties,
+  MaPointFeature,
+} from './buurt/datasets';
+import {
+  filterDatasetFeatures,
+  filterFeaturesinRadius,
+  getBboxFromFeatures,
+} from './buurt/helpers';
+import { fetchMyLocation } from './home';
+import { AppRoutes } from '../../universal/config/routes';
+import { Themas } from '../../universal/config/thema';
+import {
+  apiDependencyError,
+  apiSuccessResult,
+} from '../../universal/helpers/api';
+import { MyNotification } from '../../universal/types';
+import { AuthProfileAndToken } from '../auth/auth-types';
+
+const ADULT_AGE = 18;
+const LATE_TEEN_AGE = 16;
+const WITHIN_RADIUS_KM = 0.1; // 100m
+const datasetId = 'afvalcontainers';
+const config = datasetEndpoints[datasetId];
+const filters = {
+  [datasetId]: {
+    geadopteerd_ind: { values: { Nee: 1 } },
+  },
+};
+
+const filterQueryParam = encodeURIComponent(JSON.stringify(filters));
+
+export async function fetchAdoptableTrashContainers(
+  requestID: RequestID,
+  authProfileAndToken: AuthProfileAndToken
+) {
+  const BRP = await fetchBRP(requestID, authProfileAndToken);
+
+  if (!BRP.content?.persoon?.geboortedatum) {
+    return apiDependencyError({ BRP });
+  }
+
+  const age = differenceInYears(
+    new Date(),
+    BRP.content?.persoon?.geboortedatum
+  );
+  if (age < LATE_TEEN_AGE) {
+    return apiSuccessResult({
+      tips: [],
+    });
+  }
+
+  const MY_LOCATION = await fetchMyLocation(requestID, authProfileAndToken);
+
+  if (MY_LOCATION.status !== 'OK' || !MY_LOCATION.content?.[0]?.latlng) {
+    return apiDependencyError({ MY_LOCATION });
+  }
+
+  const afvalcontainersDatasetResponse = await fetchDataset(
+    requestID,
+    datasetId,
+    config,
+    {}
+  ).then((result) => {
+    return {
+      ...result,
+      id: datasetId,
+    };
+  });
+
+  if (afvalcontainersDatasetResponse.status !== 'OK') {
+    return apiDependencyError({
+      AFVALCONTAINERS: afvalcontainersDatasetResponse,
+    });
+  }
+
+  const latlng = MY_LOCATION.content[0].latlng;
+
+  const DO_SWAP_LAT_LNG_FOR_DISTANCE_COMPARISON = true;
```

Comment:
```suggestion
  // Incoming Coordinates are in a different format (lng,lat instead of lat,lng) than we use for comparison.
  const DO_SWAP_LAT_LNG_FOR_DISTANCE_COMPARISON = true;
```